src/main/java/net/codejava/MvcController.java

```diff
@@ -522,4 +540,180 @@ public String submitDispute(@ModelAttribute("user") User user) {
+    String userPassword = jdbcTemplate.queryForObject(getUserPasswordSql, String.class);
+
+    // unsuccessful login
+    if (userPasswordAttempt.equals(userPassword) == false) {
+      return "welcome";
+    }
+
+
+    double userWithdrawAmt = user.getAmountToWithdraw();
+    int userWithdrawAmtInPennies = (int) (userWithdrawAmt * 100);
+    
+    if (userWithdrawAmt < 0){
+      return "welcome";
+    }
+
+    // how many times to apply interest
+    int interestCount = 0;    
+
+    // check if date of the savings input passed seven days, and if amounts are equal
+    System.out.println( "WITHDRAW AMOUNT: " + userWithdrawAmtInPennies);
+    String getSavingsAmount = String.format("SELECT Amount FROM savings WHERE CustomerID='%s' AND Amount = %d;", 
+    userID, userWithdrawAmtInPennies);
+    int savingsAmount = jdbcTemplate.queryForObject(getSavingsAmount, Integer.class);
+
+    String getSavingsDate = String.format("SELECT Timestamp FROM savings WHERE CustomerID='%s' AND Amount = %d;", 
+    userID, userWithdrawAmtInPennies);
+    String savingsDate = jdbcTemplate.queryForObject(getSavingsDate, String.class);
+
+
+    // if the withdraw request matches the savings deposit
+    if(savingsAmount == userWithdrawAmtInPennies){
+      // parsing the numbers to create calendar objects for comparison
+      int savingsYear = Integer.parseInt(savingsDate.substring(0,4));
+      int savingsMonth = Integer.parseInt(savingsDate.substring(5,7));
+      int savingsDay = Integer.parseInt(savingsDate.substring(8,10));
+      int currYear = Integer.parseInt(SQL_DATETIME_FORMAT.substring(0,4));
+      int currMonth = Integer.parseInt((SQL_DATETIME_FORMAT.substring(5,7)));
+      int currDay = Integer.parseInt(SQL_DATETIME_FORMAT.substring(8,10));
+
+      Calendar savingsCal = new  GregorianCalendar(savingsYear, savingsMonth, savingsDay);
+      Calendar currCal = new  GregorianCalendar(currYear, currMonth, currDay);
+      
+      long numDays = ChronoUnit.DAYS.between(savingsCal.getTime().toInstant(), currCal.getTime().toInstant());
+      System.out.println(numDays);
+      // cannot obtain seconds between calender dates, for testing I simply the interest count to 2
+      Random r = new Random(); // used for testing
+      interestCount+=2;
+      // time is less than a week
+      if(interestCount == 0){
+        return "welcome_page";
+      }
+      depositAmount = savingsAmount;
+      // if the customer has not withdrawn in a long time, the interest will be applied numerous times
+      for(int i = 0; i < interestCount; i++){
+        depositAmount *= SAVINGS_INTEREST;
+      }
+
+      int toBalance = (int) depositAmount;
+
+      String balanceIncreaseSql = String.format("UPDATE Customers SET Balance = Balance + %d WHERE CustomerID='%s';", toBalance, userID);
+      System.out.println(balanceIncreaseSql);
+      jdbcTemplate.update(balanceIncreaseSql);

```

Comment:
This logic should be wrapped in a helper function and called whenever the user interacts with their account:
- Simply logs in to check their balance(s)
- Deposits/Withdraws/Disputes
- Deposits more into Savings Account